app/services/course_creation_step_service.rb

```diff
@@ -0,0 +1,139 @@
+class CourseCreationStepService
+  def execute(current_step:, course:)
+    workflow_steps = get_workflow_steps(course)
+
+    {
+      next: get_next_step(workflow_steps, current_step),
+      previous: get_previous_step(workflow_steps, current_step),
+    }
+  end
+
+  def get_next_step(steps, current_step)
+    steps[steps.find_index(current_step).next]
+  end
+
+  def get_previous_step(steps, current_step)
+    steps[steps.find_index(current_step).pred]
+  end
+
+  def get_workflow_steps(course)
+    if course.is_further_education?
+      further_education_workflow_steps(course)
+    elsif course.is_uni_or_scitt?
+      uni_or_scitt_workflow_steps(course)
+    elsif course.is_school_direct?
+      school_direct_workflow_steps(course)
+    end
+  end
+
+  def school_direct_workflow_steps(course)
+    if course.provider.sites && course.provider.sites.count > 1
+      school_direct_workflow_with_locations_steps
+    else
+      school_direct_workflow_without_locations_steps
+    end
```

Comment:
Related to above - I find this duplication of all the steps quite hard to follow, and it will probably be a pain if we ever want to alter any workflow

Could you instead get the full workflow, and remove `:locations` from it, explicitly showing that we remove it from the workflow?

Reply:
Follow up - Locations does actually do something when you hit it, it assigns the site ID with the only one it finds. This appears to break that behaviour as it now skips that logic (Side note, we should have a test that covers this)
 
See: https://github.com/DFE-Digital/manage-courses-frontend/blob/master/app/controllers/courses/sites_controller.rb#L10-L15